### PR TITLE
#97 - Vue-route-guard

### LINKED ISSUE #97: Investigate adding a vue demoapp and ability to generate vue libraries to the repo
As a vue developer, I'd like to be able to create libraries that would help vue developers.

**Note:** https://www.npmjs.com/package/@nx-plus/vue might be the nx plugin we need.

### PR DESCRIPTION
**Pull request by temitopeadesoji**
## Bug

- [x] The code is properly formatted using the project's prettier and eslint settings
- [x] Make sure everything is properly typed in the code.

## Feature

fixes #97 
- Added vue to nx setup using `nx-vue3-vite`
- Gives the ability for users to setupGuard
- Gives the ability for users to injectGuard into a vue page

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

File: libs/vue-route-guard/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "strict": true,
+    "allowJs": true,
```
Comment (ktrz) [inline libs/vue-route-guard/tsconfig.json:5]: why `allowJs` is `true`?

File: package.json [hunk 2]
```diff
@@ -54,59 +55,87 @@
...
     "@angular/cli": "~14.0.0",
     "@angular/compiler-cli": "14.0.2",
     "@angular/language-service": "14.0.2",
+    "@babel/core": "^7.18.5",
+    "@babel/plugin-transform-modules-commonjs": "^7.18.2",
+    "@babel/preset-env": "^7.18.2",
     "@commitlint/cli": "^12.1.1",
     "@commitlint/config-conventional": "^12.1.1",
+    "@cypress/code-coverage": "^3.10.0",
+    "@cypress/vite-dev-server": "^3.0.0",
```
Comment (ktrz) [inline package.json:82]: I think those deps are causing resolution conflicts with nx cypress version

File: package.json [hunk 3]
```diff
@@ -54,59 +55,87 @@
...
+    "@cypress/vue": "^4.0.0",
+    "@intlify/eslint-plugin-vue-i18n": "^2.0.0",
     "@jscutlery/semver": "^2.23.4",
     "@nrwl/cli": "14.3.5",
-    "@nrwl/cypress": "14.3.5",
+    "@nrwl/cypress": "*",
```
Comment (ktrz) [inline package.json:87]: this doesn't seem like a good idea

File: package.json [hunk 4]
```diff
@@ -11,35 +11,36 @@
...
-    "start": "nx serve showcase",
-    "build": "nx build",
-    "test": "nx test",
-    "lint": "nx workspace-lint && nx lint",
-    "lint:fix": "nx workspace-lint && nx lint --fix",
-    "e2e:chrome": "nx e2e showcase-e2e --browser=chrome",
-    "e2e:edge": "nx e2e showcase-e2e --browser=edge",
-    "e2e:ff": "nx e2e showcase-e2e --browser=firefox",
-    "e2e:debug": "nx e2e showcase-e2e --watch",
+    "affected": "nx affected",
```
Comment (ktrz) [inline package.json:14]: why did you reorder all the scripts? it's hard to track what has changed

**Review by ktrz — COMMENTED** [review]
added some comments

I think `libs/.gitkeep` file is not necessary and can be removed

File: libs/vue-route-guard/tsconfig.json (hunk 1 above)
Comment (temitopeadesoji) [reply libs/vue-route-guard/tsconfig.json:5]: i will remove it, it was part of the default setting configured by the package i used to setup the project

File: package.json (hunk 4 above)
Comment (temitopeadesoji) [reply package.json:14]: `npm audit` rearranged them alphabetically

File: .vscode/extensions.json [hunk 5]
```diff
@@ -3,6 +3,8 @@
     "angular.ng-template",
     "nrwl.angular-console",
     "esbenp.prettier-vscode",
-    "firsttris.vscode-jest-runner"
+    "firsttris.vscode-jest-runner",
+    "johnsoncodehk.volar",
```
[the remark this answers is not in the document]
Comment (temitopeadesoji) [reply .vscode/extensions.json:7]: The package i used added it to the setup cause volar is the default vue code formmatter for vue 3

File: package.json (hunk 3 above)
Comment (temitopeadesoji) [reply package.json:87]: Yes but i am not exactly sure how to resolve the deps that are caused

File: package.json (hunk 2 above)
Comment (temitopeadesoji) [reply package.json:82]: yes, they are. do you have any suggestions on how to resolve this

File: package.json (hunk 4 above)
Comment (temitopeadesoji) [reply package.json:14]: do you suggest i revert them back to how it previously was? @ktrz

Comment (temitopeadesoji) [reply package.json]: reverted

**Review by ktrz — APPROVED** [review]

**Pull request merged**